client/index.js

```diff
@@ -10,13 +10,15 @@
     document.querySelector('#messages').appendChild(logEntry);
   };
 
-  const checkUnlocked = () =>
-    new Promise((resolve, reject) => {
+  const checkUnlocked = async () => {
+    await window.ethereum.enable() // Ensure access to MetaMask
```

Comment:
I feel like we should check for the existence of the ethereum object first.

Reply:
Good point. Also changed to use the localised window alias `w`